### PR TITLE
server/coordinator: startup schedulers with considering the proportion of regions

### PR DESCRIPTION
**Pull request by nolouch**
…n of regions activated on each store

### What problem does this PR solve? 
If we have many stores, like 10 stores, and restart the cluster. 9 stores already reported a heartbeat and the last not finished. the schedulers will begin to schedule and do some unnecessary balance.
### What is changed and how it works?
This PR startup schedulers with considering the proportion of regions activated on each store.
### Check List 

Tests 
 - Unit test

### CODE REVIEW

**Comment by disksing** [comment]
em, have you run a cluster to manually test it?

File: server/core/region.go [hunk 1]
```diff
@@ -486,10 +486,6 @@ func (r *RegionsInfo) AddRegion(region *RegionInfo) []*metapb.Region {
 
 	r.regions.Put(region)
 
-	if region.Leader == nil {
```
[the remark this answers is not in the document]
Comment (nolouch) [reply server/core/region.go]: It not illegal.

**Comment by disksing** [comment]
@nolouch @rleungx Can we push this PR forward now?

**Comment by nolouch** [comment]
I do 2 case with the manual test. One is simulator 200 stores and Isolate the network of PD and TiKV, then restart the PD and the coordinator will not start until timeout. Another with 4 stores and isolate a store then restart the PD, the coordinator will start because the heartbeat can be completely collected by the new leaders.

**Pull request merged**